resource_apiscope.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/3lvia/terraform-provider-elvid/elvidapiclient"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceApiScope() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceApiScopeCreateOrUpdate,
+		ReadContext:   resourceApiScopeRead,
+		DeleteContext: resourceApiScopeDelete,
+		UpdateContext: resourceApiScopeCreateOrUpdate,
+		Schema: map[string]*schema.Schema{
+			// The id field has to be Optional and Computed. So we have the name field in addition to the id field even if they'll have the same value.
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the API scope",
```

Comment:
Vi bør ha ForceNew: true for name.

Slik at terraform skjønner at dette blir en ny ressurs ved endring. 

Slik koden i elvid er nå vil det faktisk bli laget et nytt scope per nå, og det gamle blir liggende igjen.

Reply:
Godt poeng 👍